core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -338,7 +338,11 @@ public String name() {
 
     @Override
     public void applyTo(TableMetadata.Builder metadataBuilder) {
-      metadataBuilder.removeRef(refName);
+      if (refName.equals(SnapshotRef.MAIN_BRANCH)) {
+        metadataBuilder.resetMainBranch();
+      } else {
+        metadataBuilder.removeRef(refName);
+      }
```

Comment:
Yeah at this point we may as well make TableMetadata.removeRef not clear the log if it's main, and just have the consistent behavior for resetting to main. I feel like that's the sane default for that API now but the only issue is if others depend on that behavior